tests/acceptance/features/apiSpacesShares/shareSpacesViaLink.feature

```diff
@@ -128,3 +128,66 @@ Feature: Share spaces via link
     Then the HTTP status code should be "400"
     And the OCS status message should be "Can not share space root"
     And for user "Alice" the space "Alice Hansen" should not contain the last created public link
+
+
+  Scenario: space admin removes password of a public link share (read permission) of a space
+    Given user "Alice" has created a public link share of the space "share space" with settings:
+      | permissions | 1        |
+      | password    | %public% |
+    When user "Alice" updates the last public link share using the sharing API with
+      | permissions | 1 |
+      | password    |   |
+    Then the HTTP status code should be "200"
+    And the OCS status code should be "100"
+
+
+  Scenario Outline: space admin tries to remove password of a public link space share with various permissions
+    Given user "Alice" has created a public link share of the space "share space" with settings:
+      | permissions | <permissions> |
+      | password    | %public%      |
+    When user "Alice" updates the last public link share using the sharing API with
+      | permissions | <permissions> |
+      | password    |               |
+    Then the HTTP status code should be "200"
+    And the OCS status code should be "400"
+    And the OCS status message should be "missing required password"
+    Examples:
+      | permissions |
+      | 5           |
+      | 15          |
+      | 4           |
+
+
+  Scenario: space manager tries to remove password of a public link space share (read permission)
+    Given user "Alice" has shared a space "share space" with settings:
+      | shareWith | Brian   |
+      | role      | manager |
+    And user "Brian" has created a public link share of the space "share space" with settings:
+      | permissions | 1        |
+      | password    | %public% |
+    When user "Brian" updates the last public link share using the sharing API with
+      | permissions | 1 |
+      | password    |   |
+    Then the HTTP status code should be "200"
+    And the OCS status code should be "104"
+    And the OCS status message should be "user is not allowed to delete the password from the public link"
```

Comment:
This is a peculiar case where the error message and OCS status code is different. If I'm not wrong `1xx` is the success for `ocs` status code? The error message on this one is however very nice. I have written this scenario as the server  behaves right now as I don't know if this is an expected response or not.
Should the error code be `400` like in other similar cases what about the error message? 
cc @ScharfViktor @saw-jan

Reply:
Let's leave it at that. The main thing is that Brian can't delete the password. The web hides the password delete button for Brian, so the user can only try to delete the password using the api (he can see the message there).